lib/setupHooks/removeReferencesToVendoredSourcesHook.sh

```diff
@@ -26,13 +26,23 @@ removeReferencesToVendoredSources() {
 
       echo -n '\)!@storeDir@/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee!g'
     )
+
+    # Sign modified binaries on aarch64-darwin
+    if [ -n "${signingUtils}" ]; then
+      signIfRequired "${installedFile}"
+    fi
   done < <(find "${installLocation}" -type f)
 }
 
 if [ -n "${doNotRemoveReferencesToVendorDir-}" ]; then
   echo "removeReferencesToVendoredSources disabled"
 elif [ -n "${cargoVendorDir-}" ]; then
-  postInstallHooks+=(removeReferencesToVendoredSources)
+  signingUtils="@signingUtils@"
+  if [ -n "${signingUtils}" ]; then
+    source "${signingUtils}"
+  fi
```

Comment:
Couple comments:

1. We should unconditionally source this file and always have the functions available (I peeked at the sources, there's no automatic behavior just function definitions). One way we can avoid the checks completely is use `@sourceSigningUtils@` in here and in the derivation definition we can conditionally replace that string with `"source ${signingUtils}`
2. That source should be done outside of these conditional checks (that way if someone manually calls `removeReferencesToVendoredSources` without `$cargoVendorDir` being set it won't crash)

Reply:
I don't think always sourcing `signingUtils` is a good idea, unless it's replaced with an empty file on non-aarch64-darwin system. It will pull in darwin specific binaries as build dependencies.

Reply:
`@sourceSigningUtils@` is a good idea though.